src/proto/stream.rs

```diff
@@ -161,18 +162,17 @@ where
                 } else {
                     self.state = StreamState::ClosedByPeer;
 
-                    if !self.frame_queue.iter().any(EncodedFrame::is_close) {
-                        match &e {
-                            Error::Protocol(e) => self.queue_frame(Frame::from(e)),
-                            Error::PayloadTooLong { max_len, .. } => self.queue_frame(
-                                Message::close(
-                                    Some(CloseCode::MESSAGE_TOO_BIG),
-                                    &format!("max length: {max_len}"),
-                                )
-                                .into(),
-                            ),
-                            _ => {}
-                        }
+                    // The state was not ClosedByUs, so we did not send a close frame yet.
```

Comment:
Feels redundant, but if you want to, this could be replaced by `debug_assert_eq!(self.state, StreamState::ClosedByPeer)` at the start

Reply:
Or if coments like these could be moved to the `ClosedByUs` variant definition instead of being repeated inline

Reply:
I do agree that the comments are entirely unnecessary, the variant names should be self-explanatory